include/envoy/upstream/locality_endpoint.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "envoy/config/core/v3/base.pb.h"
+#include "envoy/config/endpoint/v3/endpoint_components.pb.h"
+
+#include "common/protobuf/utility.h"
+
+namespace Envoy {
+namespace Upstream {
+
+using LocalityEndpointTuple = std::tuple<const envoy::config::core::v3::Locality&,
```

Comment:
Does this type appear in pure interfaces? If it's only internal to HDS, I suggest placing the file in `source/common/upstream`.

Reply:
I moved the new Equal and Hash structs for HC and Endpoints to Upstream, as you mention since they are only used within HDS.